source/core/AgentBase.hpp

```diff
@@ -39,6 +42,44 @@ namespace cse491 {
     bool IsAgent() const override { return true; }
 
 
+    /// Insert pair of item_name and num_uses, or update num_uses if possible
+    /// If num_uses == 0, removes the pair from inventory_map
+    void UpdateInventory(const std::string& item_name, const size_t& num_uses) {
```

Comment:
When you pass by reference, you're effectively passing a pointer into the function instead of the full value.  For larger objects (like many strings!) this can be a helpful timesaver.  If smaller values like size_t that you aren't changing, it's best to just pass by value.